File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/converter/UserDTOMixin.java
package com.sv.youapp.component.authorization.converter;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.security.core.GrantedAuthority;

import java.time.Instant;
import java.util.Set;

public abstract class UserDTOMixin {

    @JsonProperty("id")
    Integer id;

    @JsonProperty("username")
    String username;

    @JsonProperty("password")
    String password;

    @JsonProperty("email")
    String email;

    @JsonProperty("profilePictureUrl")
    String profilePictureUrl;

    @JsonProperty("registeredAt")
    Instant registeredAt;

    @JsonProperty("authorities")
    Set<GrantedAuthority> authorities;

    @JsonProperty("isEnabled")
    Boolean enabled;
}

File: authorization-server/src/main/kotlin/com/sv/youapp/authorization/converter/UserDTOMixin.kt
package com.sv.youapp.authorization.converter

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.security.core.GrantedAuthority
import java.time.Instant

abstract class UserDTOMixin(
    @JsonProperty("id") val id: Int,
    @JsonProperty("username") val username: String,
    @JsonProperty("password") val password: String,
    @JsonProperty("email") val email: String,
    @JsonProperty("profilePictureUrl") val profilePictureUrl: String?,
    @JsonProperty("registeredAt") val registeredAt: Instant,
    @JsonProperty("authorities") val authorities: Set<GrantedAuthority>,
    @JsonProperty("isEnabled") val enabled: Boolean,
)
